schema/atop_xslt.sch

```diff
@@ -13,63 +13,82 @@
     <!-- Global variables. -->
     <let name="docUri" value="document-uri(/)"/>
     
+    <let name="atopNamespaceUri" value="'http://www.tei-c.org/ns/atop'"/>
+    <let name="atopNamespacePrefix" value="'atop'"/>
+    
     <let name="reVarName" value="'^v[A-Z][a-zA-Z0-9]+(_[a-zA-Z0-9_]+)?$'"/>
     
     <let name="reParamName" value="'^t?p[A-Z][a-zA-Z0-9]+(_[a-zA-Z0-9_]+)?$'"/>
     
-    <let name="reFunctionName" value="'^teif:[a-z][a-z\-]+[a-z]$'"/>
+    <let name="reFunctionName" value="concat('^', $atopNamespacePrefix, ':[a-z][a-z\-]+[a-z]$')"/>
     
     <!-- Constraints -->
-    
-    <pattern id="things-must-have-as-attribute">
-        <rule context="xsl:template | xsl:variable | xsl:with-param | xsl:param | xsl:function">
-            <let name="ln" value="local-name(.)"/>
-            <let name="name" value="@name"/>
-            <assert test="@as">
-                ERROR: <value-of select="$name"/> (<value-of select="$ln"/>) missing required @as attribute.
+    <pattern id="names">
+        <title>Coding style rules for names</title>
+        <rule abstract="true" id="param-name-re">
+            <assert test="matches(tokenize(@name, ':')[last()], $reParamName)" id="assert-param-name-re" role="error">
+                ERROR: A <value-of select="name()"/> must have a name matching the regular expression <value-of select="$reParamName"/>. 
             </assert>
         </rule>
-    </pattern>
-    
-    <pattern id="variable-names">
+        <rule abstract="true" id="variable-name-re">
+            <assert test="matches(tokenize(@name, ':')[last()], $reVarName)" id="assert-variable-name-re" role="error">
+                ERROR: A <value-of select="name()"/> must have a name matching the regular expression <value-of select="$reVarName"/>. 
+            </assert>
+        </rule>
+        <rule abstract="true" id="global-name">
+            <assert test="starts-with(@name, concat($atopNamespacePrefix, ':'))" role="error" id="assert-global-name-prefix">
+                ERROR: The name of a global <value-of select="name()"/> must be a xs:QName with the prefix <value-of select="$atopNamespacePrefix"/>. 
+            </assert>
+            <assert test="namespace-uri-for-prefix(substring-before(@name, ':'), .) = $atopNamespaceUri" role="error" id="assert-global-name-uri">
+                ERROR: The name of a global <value-of select="name()"/> must be a xs:QName with the namespace URI <value-of select="$atopNamespaceUri"/>. 
+            </assert>
+        </rule>
+        <rule abstract="true" id="local-name">
+            <assert test="not(contains(@name, ':'))" role="error" id="assert-local-name">
```

Comment:
Please, `<report test="contains( @name,':') id="report-local-name">`. My head was spinning.

Actually, that is insufficient, though, isn’t it? Or has regular schema-checking already tested that the value of `@name` is an xs:Name? I think it has. If you think it has not, we could just use `<assert test="@name castable as xs:NCName">`.

Reply:
Changed in [f463784](https://github.com/TEIC/atop/pull/5/commits/f463784acd6b025b04102dde69f6d5a11273ea77) to castable as xs:NCName